lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexUserCode.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+/**
+ * The starting point of Vortex user thread.
+ */
+public interface VortexUserCode {
+  /**
+   * Implement this method, using VortexThreadPool to run jobs on Vortex.
+   * Your implementation of this interface will be instantiated by Tang
+   * and this method will be called upon the start of REEF Driver.
+   */
+  void start(final VortexThreadPool vortexThreadPool);
```

Comment:
I guess if we only allow a single VortexThreadPool instance, then the current design seems reasonable, or maybe we could use static methods or singleton instead?

Another possibility is to allow dynamic creation of VortexThreadPool instances, each of which manages a disjoint set of Executors. And in that case, we might just let user to do whatever he wants; but I'm not sure if this is something easy to support in a REEF Driver.

Zhengping, from my phone

> On Jul 24, 2015, at 8:15 AM, John Yang <notifications@github.com> wrote:
> 
> In lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexUserCode.java:
> 
> > + * KIND, either express or implied.  See the License for the
> > + * specific language governing permissions and limitations
> > + * under the License.
> > + */
> > +package org.apache.reef.vortex.api;
> > +
> > +/**
> > + * The starting point of Vortex user thread.
> > + */
> > +public interface VortexUserCode {
> > +  /**
> > +   * Implement this method, using VortexThreadPool to run jobs on Vortex.
> > +   * Your implementation of this interface will be instantiated by Tang
> > +   * and this method will be called upon the start of REEF Driver.
> > +   */
> > +  void start(final VortexThreadPool vortexThreadPool);
> @zpqian and I wanted to make it explicit for the Vortex users to use VortexThreadPool injected by Tang. Currently we inject VortexThreadPool in Vortex internal code and pass it to VortexuserCode via this method.
> 
> Then do you prefer the user by herself inject VortexThreadPool into the constructor of her implementation of VortexUserCode and use it inside start()? Or maybe should we somehow rethink the overall api design?
> 
> @zpqian, what are your thoughts?
> 
> —
> Reply to this email directly or view it on GitHub.
>